client/concordclient/src/concord_client.cpp

```diff
@@ -31,6 +35,75 @@ ConcordClient::ConcordClient(const ConcordClientConfig& config)
   auto trc_config = std::make_unique<ThinReplicaClientConfig>(
       config_.subscribe_config.id, trc_queue_, config_.topology.f_val, std::move(trs_connections));
   trc_ = std::make_unique<ThinReplicaClient>(std::move(trc_config), metrics_);
+  ConcordClientPoolConfig client_pool_config = std::move(createClientPoolStruct(config));
+  client_pool_ = std::make_unique<concord::concord_client_pool::ConcordClientPool>(client_pool_config, metrics_);
+  if (client_pool_->HealthStatus() == concord::concord_client_pool::PoolStatus::NotServing) {
+    LOG_ERROR(KEY_EX_LOG, "Client pool health status is Not Serving");
+    throw concord::concord_client_pool::InternalError();
+  }
+}
+
+ConcordClientPoolConfig& ConcordClient::createClientPoolStruct(const ConcordClientConfig& config) {
+  ConcordClientPoolConfig* client_pool_config = new ConcordClientPoolConfig;
```

Comment:
No need to deal with the heap here. We can simply create the object on the stack and return it by value.
```
ConcordClientPoolConfig ConcordClient::createClientPoolConfig(const ConcordClientConfig& config) {
  ConcordClientPoolConfig client_pool_config;
```

Reply:
changed it, thanks.